tests/TestCase/ORM/TableTest.php

```diff
@@ -5839,6 +5839,24 @@ public function testFindOrCreateNoTransaction()
         $this->assertEquals('Success', $article->title);
     }
 
+    /**
+     * Test that findOrCreate creates a new valid entity with the search data
+     *
+     * @return void
+     */
+    public function testFindOrCreateValidatesNewEntity()
+    {
+        $articles = $this->getTableLocator()->get('Articles');
+        $validator = new Validator();
+        $validator->notBlank('title');
+        $articles->setValidator('default', $validator);
+
+        $article = $articles->findOrCreate(['title' => '']);
+        $this->assertTrue($article->isNew());
```

Comment:
The output of   `findOrCreate` should be a valid entity on DB , or `false`/`exception`  otherwise imo
so this `$this->assertTrue($article->isNew());` should not happened

Reply:
I would prefer an exception over `false`. Returning false makes our typehints hard and isn't as clear as an exception would be.

Reply:
The exception could 'break' application code, but currently that code is broken in a more serious way. Returning an unsaved entity goes against the goal/purpose of this method.

Reply:
I would prefer an exception as well, like saveOrFail, but how could the user retrieve the invalid entity to possibly do work on? Is getting the entity useful after a failed save?

Reply:
It could help in debugging/logging. We could create a new exception which contains the entity.